### DISCUSSION TITLE
SEVERE PROBLEM: Is it okay for me to use beta versions of web browsers with the Internet Archive Downloader extension?

### DISCUSSION THREAD

**Discussion by AnxiousPleaseHelp** [elementdavv/internet_archive_downloader#64] [General]
Hello,

I just want to know if it's safe to use beta versions of browsers with the Internet Archive Downloader extension? For example, the Beta version of Microsoft Edge Beta or the Beta version of Google Chrome Beta while using your extension. What about forks of Firefox like Waterfox? Is your extension compatible with these browsers?

The Microsoft Edge Beta version located here: https://www.microsoft.com/en-us/edge/download/insider?form=MA13FJ

Or Google Chrome Beta located here: https://www.google.com/intl/en_au/chrome/beta/

Will I be missing any pages from the ebooks I download from archive.org with your extension? 

This problem has really stopped me from making progress so any help would be great... 😢 😢 😢 😢 😢 😢

**Comment by elementdavv** [comment]
> I just want to know if it's safe to use beta versions of browsers with the Internet Archive Downloader extension? For example, the Beta version of Microsoft Edge Beta or the Beta version of Google Chrome Beta while using your extension. What about forks of Firefox like Waterfox? Is your extension compatible with these browsers?
> 
> The Microsoft Edge Beta version located here: https://www.microsoft.com/en-us/edge/download/insider?form=MA13FJ
> 
> Or Google Chrome Beta located here: https://www.google.com/intl/en_au/chrome/beta/
> 

There are so many tests on Edge, Firefox and several Chromium based browsers there, besides Windows, Mac OS, and Linux platforms. So I do not test with any beta version or forks of any browsers. But from the past browsers releases, the extension can work except for rare cases, eg. Firefox 133.0 which has critical changes on ServiceWorker.

> Will I be missing any pages from the ebooks I download from archive.org with your extension?
> 

No. The download will stop and failed if it failed to get any page image or text. So you can trust the resulted Pdf files.

**Comment by elementdavv** [comment]
I have just made a test on the browsers you refer, Edge beta, Chrome beta and Waterfox latest v6.5.2 in Linux, the extension just works perfectly.

**Reply by AnxiousPleaseHelp** [reply]
> I have just made a test on the browsers you refer, Edge beta, Chrome beta and Waterfox latest v6.5.2 in Linux, the extension just works perfectly.

With all these copycats ripping off your work, which are the safe browsers to use now in August 2025? 😢 😢 😢 😢

**Reply by elementdavv** [reply]
> With all these copycats ripping off your work, which are the safe browsers to use now in August 2025? 😢 😢 😢 😢

It depends on your favorites. I like open source ones.

**Reply by AnxiousPleaseHelp** [reply]
> > With all these copycats ripping off your work, which are the safe browsers to use now in August 2025? 😢 😢 😢 😢
> 
> It depends on your favorites. I like open source ones.

Is there a particular browser that you would say contains 100% of all of the ebooks content? I just worry about getting an ebook from Internet Archive that's not entirely 100% the correct ebook... 😢 😢 😢

**Reply by AnxiousPleaseHelp** [reply]
> > With all these copycats ripping off your work, which are the safe browsers to use now in August 2025? 😢 😢 😢 😢
> 
> It depends on your favorites. I like open source ones.

Would you say Microsoft Edge (Not the beta or the developer or the canary version but just the original Microsoft Edge) is the best browser to use with your extension as you mentioned it doesn't require any tweaks like Firefox does?

**Reply by AnxiousPleaseHelp** [reply]
> It depends on your favorites. I like open source ones.

What do you recommend as the best method to make sure the ebook I download from Internet Archive is the correct version?

So far I've been using Edge, Brave and Firefox to download the same ebook from the Internet Archive using all three web browsers.

Is there anything else I can do to make sure I'm getting the correct ebook version? 😢😢😢

**Reply by elementdavv** [reply]
> What do you recommend as the best method to make sure the ebook I download from Internet Archive is the correct version?
> 
> So far I've been using Edge, Brave and Firefox to download the same ebook from the Internet Archive using all three web browsers.
> 
> Is there anything else I can do to make sure I'm getting the correct ebook version? 😢😢😢

I don't know what you mean. Are the above three browsers not enough for you?
